src/plugins/vis_type_timeseries/public/application/visualizations/views/timeseries/index.js

```diff
@@ -73,16 +73,21 @@ export const TimeSeries = ({
   const chartRef = useRef();
 
   useEffect(() => {
-    const updateCursor = (_, cursor) => {
-      if (chartRef.current) {
+    const updateCursor = ({ eventName, cursor }) => {
+      if (
+        chartRef.current &&
+        eventName === ACTIVE_CURSOR &&
+        cursor &&
+        chartRef.current.chartStore.getState().chartId !== cursor.chartId
+      ) {
         chartRef.current.dispatchExternalPointerEvent(cursor);
       }
     };
 
-    eventBus.on(ACTIVE_CURSOR, updateCursor);
+    const subscription = eventBus.asObservable().subscribe(updateCursor);
```

Comment:
Could you please explain why you added `asObservable()`?

Reply:
It's a my misunderstanding. I changed it and use subscribe straight from Subject.